### ISSUE TITLE
False positive in RuboCop::Cop::Style::HashEachMethods

### ISSUE THREAD

**Issue by alex-damian-negru** [rubocop-hq/rubocop#9292]
I am using [dry-types](https://dry-rb.org/gems/dry-types/1.2/enum/), which allows me to build enum objects, such as this:

```ruby
Level = Dry::Types::String.enum('basic', 'active', 'advanced')
```

The resulting object has a `#values` method which yields an `Array` containing the values above. 

Using `Level.values.each` triggers this cop, even though I'm not operating on a hash.

--------

## Expected behavior

No offense should be added when using `#values` on objects other than those belonging to `Hash`.

## Actual behavior

Offense was added for a class different from `Hash`.

## Steps to reproduce the problem

This should do the trick:

```ruby
class Whatever
  def initialize(*items)
    @items = items
  end

  def values
    @items
  end
end

Whatever.new('a', 'b', 'c').values.each { |item| p item }
```

## RuboCop version

```
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.6.6 x86_64-linux-musl)
  - rubocop-performance 1.9.1
  - rubocop-rspec 2.1.0
```

**Comment by tejasbubane** [comment]
Rubocop's static analysis has limitations - it doesn't know what type of value a variable refers to at runtime. Which is one of the reasons this cop is marked as `unsafe`.

I think you have the following options:
1. Disable the cop inline, in the file, or entirely in the project. [ref](https://docs.rubocop.org/rubocop/configuration.html)
2. Run rubocop using `--safe` option to show only the offenses generated by cops which are marked as safe. If you are autocorrecting, use `rubocop -a` (lowercase `a`) to execute autocorrects only for cops marked as safe.
